web/modules/custom/owner/owner.module

```diff
@@ -57,23 +84,127 @@ function owner_inline_entity_form_entity_builder($entity_type, EntityInterface $
+  /** @var \Drupal\rdf_entity\RdfInterface $entity */
+  $entity = $form_state->getFormObject()->getEntity();
+  if ($entity->bundle() !== 'owner') {
+    return;
+  }
+
+  $form['#validate'][] = 'owner_form_alter_validate';
+}
+
+/**
+ * Form validation callback.
+ *
+ * Checks that an owner entity cannot transition to the 'Request deletion' state
+ * if it is still associated with a collection.
+ *
+ * @see owner_form_rdf_entity_form_alter()
+ */
+function owner_form_alter_validate(&$form, FormStateInterface $form_state) {
+  /** @var \Drupal\rdf_entity\RdfInterface $entity */
+  $entity = $form_state->getFormObject()->getEntity();
+
+  $element = $form_state->getTriggeringElement();
+  if (
+    // Check if the form was submitted using one of the workflow buttons.
+    !empty($element['#state_field']) &&
+    !empty($element['#state_id']) &&
+    // Check if we are transitioning to "Request deletion".
+    $element['#state_id'] === 'deletion_request' &&
+    // Check if the owner is still referenced by one or more collections.
+    owner_is_owner_referenced($entity)
+  ) {
+    // Compile a list of collections that are referencing the owner, to show in
+    // the form validation error message.
+    $collection_labels = array_map(function (RdfInterface $collection) {
+      return '"' . $collection->label() . '"';
+    }, owner_get_references($entity));
+
+    $form_state->setError($element, new TranslatableMarkup('The owner cannot be deleted since it owns the following collections: @collections. Please set a different owner for these collections before requesting deletion.', [
+      '@collections' => implode(', ', $collection_labels),
+    ]));
+  }
+}
+
+/**
  * Generates the entity label from the name field for owner entities.
  *
  * @param EntityInterface $entity
  *   The entity to process.
  */
 function owner_entity_auto_label(EntityInterface $entity) {
-  $fields = [
-    'owner' => 'field_owner_name',
-  ];
-
-  // Handle only our entity bundles.
-  if (!isset($fields[$entity->bundle()])) {
+  if ($entity->bundle() !== 'owner') {
     return;
   }
 
   // Generate the label using the values of the name field.
-  $field = $fields[$entity->bundle()];
-  $names = array_column($entity->get($field)->getValue(), 'value');
+  $names = array_column($entity->get('field_owner_name')->getValue(), 'value');
   $entity->set('label', implode(', ', $names));
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_access().
+ */
+function owner_rdf_entity_access(EntityInterface $entity, $operation, AccountInterface $account) {
+  if ($entity->bundle() != 'owner') {
+    return AccessResult::neutral();
+  }
+
+  // By specifications, deletion should be forbidden if the owner entity is
+  // referenced by any other entity.
+  if ($operation === 'delete') {
+    return AccessResult::forbiddenIf(owner_is_owner_referenced($entity));
+  }
+
+  if ($account->hasPermission('administer rdf entity')) {
```

Comment:
I think @brummbar meant here that they should have access. After all this `administer rdf entity` permission is something that is granted to people that even have the power to create, update and delete RDF entity bundles.